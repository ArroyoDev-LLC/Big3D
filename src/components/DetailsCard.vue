<script lang="ts">
import { defineComponent } from 'vue'
export default defineComponent({
  name: 'DetailsCard',
  props: {
    title: {
      type: String,
      default: 'Title'
    },
    details: {
      type: String,
      default: 'Details'
    },
    img: {
      type: String,
      required: false
    }
  }
})
</script>

<template>
  <div class="bg-black text-white text-left p-4">
    <slot name="header">
      <slot name="title">
        <div>{{ title }}</div>
      </slot>
      <slot name="image">
        <img v-if="img" :src="img" class="w-full" />
      </slot>
      <slot name="details">
        <div>{{ details }}</div>
      </slot>
    </slot>
    <slot name="body" />
    <slot name="footer" />
  </div>
</template>
