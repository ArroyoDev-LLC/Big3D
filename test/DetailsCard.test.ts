import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import DetailsCard from '../src/components/DetailsCard.vue'

describe('DetailsCard', () => {
  it('should render', () => {
    const wrapper = mount(DetailsCard)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('render title', () => {
    const title = 'test title'
    const wrapper = mount(DetailsCard, { props: { title } })
    expect(wrapper.get('.title').text()).toBe(title)
  })
  it('render details', () => {
    const details = 'test details'
    const wrapper = mount(DetailsCard, { props: { details } })
    expect(wrapper.get('.details').text()).toBe(details)
  })
})
