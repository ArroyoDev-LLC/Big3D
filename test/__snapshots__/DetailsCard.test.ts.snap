// Vitest Snapshot v1

exports[`DetailsCard > should render 1`] = `
"<div class=\\"bg-black text-white text-left p-4\\">
  <div class=\\"title\\">Title</div>
  <!--v-if-->
  <div class=\\"details\\">Details</div>
</div>"
`;
